express-api/appDataSource.ts

```diff
@@ -0,0 +1,21 @@
+import { DataSource } from 'typeorm';
+import { CustomWinstonLogger } from './typeorm/utilities/CustomWinstonLogger';
+
+// test if we are in a continer or not. if false use localhost
+const hostname = process.env.CONTAINERIZED ? process.env.SERVICE_NAME : 'localhost';
+
+export const AppDataSource = new DataSource({
+  type: 'postgres',
+  host: hostname,
+  port: Number(parseInt(process.env.DATABASEPORT)),
+  username: process.env.POSTGRES_USER,
+  password: process.env.POSTGRES_PASSWORD,
+  database: process.env.POSTGRES_DB,
+  synchronize: true,
+  migrationsRun: false,
+  logging: true,
+  logger: new CustomWinstonLogger(true),
```

Comment:
What does `true` argument do? I can't find documentation on this.

Reply:
I was trying to find the link again but couldn't - there are cases (in stackoverflow) where not passing in 'true' means that the custom logger never initializes and then throws an error when it is used.